### PR TITLE
test framework: move to opae_std.h for test hooks

### PR DESCRIPTION
**Pull request by tswhison**
Rather than providing a stub that implements C standard library
functions and then using dlsym() to get the actual function address,
we override things like malloc() with opae_malloc(). In a non-mock
build, a passthrough shim (opae_std.c) is used. For mock builds,
we substitute in opae_mock.cpp, which calls into hooks from the
test_system singleton.

Signed-off-by: Tim Whisonant <tim.whisonant@intel.com>

### CODE REVIEW

File: binaries/fpgad/api/opae_events_api.c [hunk 1]
```diff
@@ -71,8 +72,8 @@ int opae_api_register_event(int conn_socket,
 
 STATIC void release_event_registry(api_client_event_registry *r)
 {
-	close(r->fd);
-	free(r);
+	opae_close(r->fd);
+	opae_free(r);
```
Comment (r-rojo) [inline binaries/fpgad/api/opae_events_api.c:76]: Are these resources acquired with an opae_wrapper in a different file?

File: binaries/fpgad/api/sysfs.c [hunk 2]
```diff
@@ -90,7 +91,7 @@ char *cstr_dup(const char *s)
 	char *p;
 	size_t len = strnlen(s, 8192);
 
-	p = malloc(len+1);
```
Comment (r-rojo) [inline binaries/fpgad/api/sysfs.c:93]: Seeing as we have out own version of cstr_dup, perhaps we should promote it to opae_std and wrap it? opae_str_dup?

File: binaries/fpgad/command_line.c [hunk 3]
```diff
@@ -97,7 +98,7 @@ STATIC bool cmd_register_null_gbs(struct fpgad_config *c, char *null_gbs_path)
 						&c->null_gbs[c->num_null_gbs])) {
 				LOG("failed to load NULL GBS \"%s\"\n", canon_path);
 				opae_unload_bitstream(&c->null_gbs[c->num_null_gbs]);
-				free(canon_path);
+				opae_free(canon_path);
```
Comment (r-rojo) [inline binaries/fpgad/command_line.c:101]: Probably allocated with canonicalize_path. Switch to regular free.

File: binaries/fpgad/command_line.c [hunk 4]
```diff
@@ -257,7 +258,7 @@ int cmd_canonicalize_paths(struct fpgad_config *c)
 		if (canon_path) {
 			snprintf(c->directory, sizeof(c->directory),
 					"%s/.opae", canon_path);
-			free(canon_path);
+			opae_free(canon_path);
```
Comment (r-rojo) [inline binaries/fpgad/command_line.c:261]: Ditto here about canonicalize_path...

File: binaries/fpgad/command_line.c [hunk 5]
```diff
@@ -404,7 +405,7 @@ int cmd_canonicalize_paths(struct fpgad_config *c)
 
 			}
 
-			free(canon_path);
+			opae_free(canon_path);
```
Comment (r-rojo) [inline binaries/fpgad/command_line.c:408]: Allocated in 🛶 nicalize_path? regular free🆓 ?

File: binaries/fpgad/command_line.c [hunk 6]
```diff
@@ -439,7 +440,7 @@ void cmd_destroy(struct fpgad_config *c)
 
 	for (i = 0 ; i < c->num_null_gbs ; ++i) {
 		if (c->null_gbs[i].filename)
-			free((char *)c->null_gbs[i].filename);
+			opae_free((char *)c->null_gbs[i].filename);
```
Comment (r-rojo) [inline binaries/fpgad/command_line.c:443]: Allocated in 🛶 nicalize_path? regular free🆓 ?

File: binaries/fpgad/command_line.c [hunk 7]
```diff
@@ -450,12 +451,12 @@ void cmd_destroy(struct fpgad_config *c)
 		for (i = 0 ; c->supported_devices[i].library_path ; ++i) {
 			fpgad_supported_device *d = &c->supported_devices[i];
 			if (d->library_path)
-				free((void *)d->library_path);
+				opae_free((void *)d->library_path);
```
Comment (r-rojo) [inline binaries/fpgad/command_line.c:454]: Allocated in 🛶 nicalize_path? regular free🆓 ?

File: binaries/fpgad/command_line.c [hunk 8]
```diff
@@ -450,12 +451,12 @@ void cmd_destroy(struct fpgad_config *c)
...
 			if (d->config)
-				free((void *)d->config);
+				opae_free((void *)d->config);
```
Comment (r-rojo) [inline binaries/fpgad/command_line.c:456]: Allocated in 🛶 nicalize_path? regular free🆓 ?

File: binaries/fpgad/command_line.c [hunk 9]
```diff
@@ -450,12 +451,12 @@ void cmd_destroy(struct fpgad_config *c)
...
 		}
 
-		free(c->supported_devices);
+		opae_free(c->supported_devices);
```
Comment (r-rojo) [inline binaries/fpgad/command_line.c:459]: Allocated in 🛶 nicalize_path? regular free🆓 ?

File: binaries/fpgad/config_file.c [hunk 10]
```diff
@@ -54,11 +55,11 @@ do { \
 				canon, \
 				len); \
 			c->cfgfile[len] = '\0'; \
-			free(canon); \
+			opae_free(canon); \
```
Comment (r-rojo) [inline binaries/fpgad/config_file.c:58]: Allocated in 🛶 nicalize_path? regular free🆓 ?

File: binaries/fpgad/config_file.c [hunk 11]
```diff
@@ -54,11 +55,11 @@ do { \
...
 			return 0; \
 		} \
  \
-		free(canon); \
+		opae_free(canon); \
```
Comment (r-rojo) [inline binaries/fpgad/config_file.c:62]: Allocated in 🛶 nicalize_path? regular free🆓 ?

File: binaries/fpgad/daemonize.c [hunk 12]
```diff
@@ -92,7 +93,7 @@ int daemonize(void (*hndlr)(int, siginfo_t *, void *), mode_t mask, const char *
 	// 7) Close all open file descriptors
 	fd = sysconf(_SC_OPEN_MAX);
 	while (fd >= 0)
-		close(fd--);
+		opae_close(fd--);
```
Comment (r-rojo) [inline binaries/fpgad/daemonize.c:96]: regular close?📕

File: binaries/fpgaconf/fpgaconf.c [hunk 13]
```diff
@@ -479,7 +480,7 @@ int main(int argc, char *argv[])
 	opae_unload_bitstream(&info);
 out_exit:
 	if (config.filename) {
-		free(config.filename);
+		opae_free(config.filename);
```
[the remark this answers is not in the document]
Comment (tswhison) [reply binaries/fpgaconf/fpgaconf.c:483]: I'll add opae_canonicalize_file_name().

File: binaries/fpgad/api/opae_events_api.c (hunk 1 above)
Comment (tswhison) [reply binaries/fpgad/api/opae_events_api.c:76]: The fd comes to fpgad via the special events socket.  The r pointer is from opae_malloc() in opae_api_register_event() in the same file.

File: binaries/fpgad/api/sysfs.c (hunk 2 above)
Comment (tswhison) [reply binaries/fpgad/api/sysfs.c:93]: I'll add opae_strdup().

File: binaries/fpgad/command_line.c (hunk 4 above)
Comment (tswhison) [reply binaries/fpgad/command_line.c:261]: Added opae_canonicalize_file_name().

File: binaries/fpgad/command_line.c (hunk 5 above)
Comment (tswhison) [reply binaries/fpgad/command_line.c:408]: Added opae_canonicalize_file_name().

File: binaries/fpgad/command_line.c (hunk 6 above)
Comment (tswhison) [reply binaries/fpgad/command_line.c:443]: Added opae_canonicalize_file_name().

File: binaries/fpgad/command_line.c (hunk 7 above)
Comment (tswhison) [reply binaries/fpgad/command_line.c:454]: Added opae_canonicalize_file_name().

File: binaries/fpgad/command_line.c (hunk 8 above)
Comment (tswhison) [reply binaries/fpgad/command_line.c:456]: Added opae_canonicalize_file_name().

File: binaries/fpgad/command_line.c (hunk 9 above)
Comment (tswhison) [reply binaries/fpgad/command_line.c:459]: Added opae_canonicalize_file_name().

File: binaries/fpgad/config_file.c (hunk 10 above)
Comment (tswhison) [reply binaries/fpgad/config_file.c:58]: Added opae_canonicalize_file_name().

File: binaries/fpgad/daemonize.c (hunk 12 above)
Comment (tswhison) [reply binaries/fpgad/daemonize.c:96]: I'd rather keep this for consistency.

**Review by r-rojo — APPROVED** [review]

**Pull request merged**